server.js

```diff
@@ -1,38 +1,218 @@
+		const querySortOrder = !sortOrder ? 1 : sortOrder;
+		sortQuery = { [querySortBy]: querySortOrder };
+	}
+
+	if (noFilters) {
+		const albumsData = await Album.find()
+			.sort(sortQuery)
+			.skip(skipIndex)
+			.limit(pageSize);
+		const albumsTotalCount = await Album.find()
+			.sort(sortQuery)
+			.countDocuments();
+
+		//Create an albumsdata object with additional information
+		if (albumsData.length > 0) {
+			const returnObj = {
+				albumsTotalCount: albumsTotalCount,
+				albumsReturned: albumsData.length,
+				pageSize: pageSize,
+				page: page,
+				results: albumsData,
+			};
+			res.json(returnObj);
+		} else {
+			res.status(404).send(errorMsg);
+		}
+	}
+
+	if (!noFilters) {
+		if (filterArtist) {
+			artistFilter = { artist: filterArtist };
+			filterArray.push(artistFilter);
+		}
+		if (filterYear) {
+			yearFilter = { year: filterYear };
+			filterArray.push(yearFilter);
+		}
+		if (filterGenre) {
+			genreFilter = { genre: filterGenre };
+			filterArray.push(genreFilter);
+		}
+		//Range cannot be used in combination with exact.
+		if (filterYearFrom && !filterYear) {
+			yearFromFilter = { year: { $gte: filterYearFrom } };
+			filterArray.push(yearFromFilter);
+		}
+
+		if (filterYearTo && !filterYear) {
+			yearToFilter = { year: { $lte: filterYearTo } };
+			filterArray.push(yearToFilter);
+		}
+
+		//Using collation with strength 2 to ignore case in search.
+		const albumsDataFiltered = await Album.find({ $and: filterArray })
+			.collation({ locale: "en", strength: 2 })
+			.sort(sortQuery)
+			.skip(skipIndex)
+			.limit(pageSize);
+
+		const albumsTotalMatchCount = await Album.find({ $and: filterArray })
+			.collation({ locale: "en", strength: 2 })
+			.countDocuments();
+
+		if (albumsDataFiltered.length > 0) {
+			const returnObj = {
+				albumsTotalCount: albumsTotalMatchCount,
+				albumsReturned: albumsDataFiltered.length,
+				pageSize: pageSize,
+				page: page,
+				filters: filterArray,
+				results: albumsDataFiltered,
+			};
+			res.json(returnObj);
+		} else {
+			res.status(404).send(errorMsg);
+		}
+	}
+});
+
+//Get top10 albums
+app.get("/api/albums/top10", async (req, res) => {
+	const topTenAlbums = await Album.find().sort({ number: 1 }).limit(10);
+	if (topTenAlbums.length > 0) {
+		res.json(topTenAlbums);
+	} else {
+		res.status(404).send(errorMsg);
+	}
+});
+
+//Get bottom10 albums
+app.get("/api/albums/bottom10", async (req, res) => {
+	const bottomTenAlbums = await Album.find().sort({ number: -1 }).limit(10);
+	if (bottomTenAlbums.length > 0) {
+		res.json(bottomTenAlbums);
+	} else {
+		res.status(404).send(errorMsg);
+	}
+});
+
+/*SINGLE ITEMS*/
+//Get by placement in list
+app.get("/api/albums/placement/:placement", async (req, res) => {
+	const placement = req.params.placement;
+	const singleAlbumFiltered = await Album.findOne({ number: placement });
+	if (singleAlbumFiltered.length > 0) {
+		res.json(singleAlbumFiltered);
+	} else {
+		res.status(404).send(errorMsg);
+	}
+});
```

Comment:
Finding it hard to understand what placement is. If I try `api/albums/placement/4` it will return an error message. 
If I try to add _id instead of 4 the server is loading for a long time, one way to prevent this is by using try catch, then the server will catch the error if not matching Model requirements.